### PR TITLE
Feat refactor loops

### PR DESCRIPTION
**Pull request by Fabianstw**
This should clean up the code for generating loops (did not change any logic)

Tried to improve the following points: 
- more clear variable names
- smaller functions
- separating for, forfor and while into different files
- not using (big blocks of) ternary operators anymore
- only comments if necessary  

Additional: 
- removed `no-select`

Future ToDo:
- remove the `pseudoCode`-package
- *(idea)* math expression (`a+b*c-d`) as user input and calculating this

### CODE REVIEW

File: shared/src/question-generators/time/utilsStars/utilsWhileCondition.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * This function compare two values for var1 and var2 inside a while loop
+ * it returns true if the condition is met, so the loop can continue
+ * otherwise false
+ * @param i
+ * @param j
+ * @param compare
+ */
+function compareTwoValues(i: number, j: number, compare: string) {
+  return compare === "<"
```
Comment (holgerdell) [inline shared/src/question-generators/time/utilsStars/utilsWhileCondition.ts:12]: Can you rephrase this as if..else if..etc.?

File: shared/src/question-generators/time/utilsStars/utilsWhile.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    firstVariableValue,
+    secondVariableValue,
+    endValue,
+    variableOrder,
+    compareOption,
+    endManipulation,
+    whileCaseOption,
+  } = computeStartEndVarsWhile(random)
+
+  if (firstVariableValue === -10 && secondVariableValue === -10) {
```
Comment (holgerdell) [inline shared/src/question-generators/time/utilsStars/utilsWhile.ts:51]: I think it's better to use the initial value `undefined` instead of -10 to indicate that the variable value has not been set yet.

**Comment by holgerdell** [comment]
Sonst ist mir hier nichts aufgefallen, kann glaube ich eingefädelt werden

**Comment by Fabianstw** [comment]
> Sonst ist mir hier nichts aufgefallen, kann glaube ich eingefädelt werden

Alles klar, ich lass gerne Alex und Janett gerne noch drüber gucken und merge es dann 👍

**Review by AbcAeffchen — COMMENTED** [review]
The refactoring looks fine. Just the file names are strange. The directory is called `utilsStar` and most files have the prefix `utils`. Just remove the prefixes and put everything in a subfolder or something like that.
